node/src/blockchain/blockchain_bridge.rs

```diff
@@ -350,9 +431,9 @@ impl BlockchainBridge {
         Ok(())
     }
 
-    fn handle_scan<M, F>(&mut self, handler: F, scan_type: ScanType, msg: &M)
+    fn handle_scan<M, F>(&mut self, handler: F, scan_type: ScanType, msg: M)
     where
-        F: FnOnce(&mut BlockchainBridge, &M) -> Result<(), String>,
+        F: FnOnce(&mut BlockchainBridge, M) -> Result<(), String>,
```

Comment:
To be honest, the ownership doesn't make much sense if we keep it at the tier up by one. Think with me:

Firstly, think about the Actor message Handler and what we could say about it? The best option is to have very clear content in there, ideally a single function and nothing else.
Next, what the handle is suppose to do in general?
To take a message and process it (it usually brings data along). What data is easiest to process? Well, the owned data. In other words _I would find it bad code if somebody widened the competences of the handler, trying to use the message for anything outside the actual msg processing machinery_. 

There is just one thing to my eyes which could make devs nervous. It is the fact that the arguments above don't match with the call of  `self.handle_scan(XXX_the_fn_to_process_the_msgXXX, other param, 
msg)` because technically it is another layer around the msg processing function per se. Yeah, I was thinking about it now as well as then and I do think it's worth a big concern. Look at the guts of it. 
One, it is very simple inside. Just two usages of the msg. Two, more importantly, it has to take it as generics. Think how inconvenient is to work with generics. Nobody sane would deliberately add anything here that would interact with the variable `msg`. He would have to implement a new trait as a complement to the other trait, which requires also defining a supertrait to be able to set bounds for the generic type. It's hassle and if somebody had done that there would have probably been another way to do that, being much easier.    

For these reasons I'm fine to require an owned version of it because it is completely safe to me. Also moving the ownership closer to the guts of the data processing functions is another and huge positive trade-off and somebody should have done it before me.

Reply:
The way you pointed out, that you don't want any mutation to happen inside msg, it totally makes sense.

I totally agree with your point of view. This is an exception, if we would be dealing with some sort of structure, I'd like to keep the powers at the highest level possible, so the functions should only be granted the permissions as per their actions. If a function is able to perform all it's instructions just by reading the structure, then I'd prefer references over mutation or ownership.

Since, this is an exception, I just want to appreciate your efforts for making the right choice.